src/InterfaceLimits.jl

```diff
@@ -169,20 +236,68 @@ function add_variables!(
         @variable(m, L, upper_bound = 0.0)
         vars["load"] = L
     end
-    if security
-        isnothing(c_branches) && error("c_branches must be defined")
-        @variable(m, CF[inames, get_name.(c_branches), get_name.(c_branches)])
+    if isa(security, Security)
+        @variable(
+            m,
+            CF[
+                inames,
+                get_name.(security.contingency_branches),
+                get_name.(security.contingency_branches),
+            ]
+        )
         vars["cont_flow"] = CF
     end
 
     return vars
 end
 
-function line_direction(br, ikey)
+function line_direction(br::ACBranch, ikey::Pair{String,String})
     forward = get_name(get_area(get_from(get_arc(br)))) == first(ikey) ? 1.0 : -1.0
     return forward
 end
 
+function get_flow_lims(br::ACBranch)
+    return (forward = get_rate(br), reverse = -1 * get_rate(br))
+end
+
+function get_flow_lims(br::TwoTerminalHVDCLine)
+    return (
+        forward = get_active_power_limits_to(br).max,
+        reverse = -1 * get_active_power_limits_from(br).max,
+    )
+end
+
+function get_flow_lims(br::Union{Branch,TwoTerminalVSCDCLine})
+    @warn "$(typeof(br)) not considered in interface limit calculations"
+    return (forward = Inf, reverse = -Inf)
+end
+
+function get_directional_flow_lim(br::ACBranch, ikey::Pair{String,String})
+    flow_lim = get_flow_lims(br)
+    dir = line_direction(br, ikey)
+    lim = dir == 1.0 ? flow_lim.forward : flow_lim.reverse * dir
+    return lim
+end
+
+function find_hvdc_buses(sys::System)
+    dc_br = get_available_components(TwoTerminalHVDCLine, sys)
+    from_b = get_from.(get_arc.(dc_br))
+    to_b = get_to.(get_arc.(dc_br))
+    return Set{ACBus}(union(from_b, to_b))
+end
+
+function get_hvdc_inj(b, iname, F)
```

Comment:
F variable indexed by branch name not bus name, so updating the function to grab the corresponding branch instead. Still have to figure out how to consider more than just the first TwoTerminalHVDCLine at the bus incase there are multiple